### PR TITLE
Allow validating CompletionTime by turning it into a pointer

### PR DESCRIPTION
**Pull request by kargakis**

### CODE REVIEW

**Comment by kargakis** [comment]
@cjwagner @spxtr ptal

File: prow/kube/prowjob.go [hunk 1]
```diff
@@ -93,16 +93,23 @@ type ProwJobSpec struct {
...
-	CompletionTime time.Time    `json:"completionTime,omitempty"`
+	CompletionTime *time.Time   `json:"completionTime,omitempty"`
 	State          ProwJobState `json:"state,omitempty"`
 	Description    string       `json:"description,omitempty"`
 	URL            string       `json:"url,omitempty"`
 	PodName        string       `json:"pod_name,omitempty"`
 	BuildID        string       `json:"build_id,omitempty"`
 }
 
+// TODO: Turn this into an utility in pjutil?
```
Comment (spxtr) [inline prow/kube/prowjob.go]: I think this is better as-is.

File: prow/kube/prowjob.go [hunk 2]
```diff
@@ -93,16 +93,23 @@ type ProwJobSpec struct {
...
 func (j *ProwJob) Complete() bool {
-	return !j.Status.CompletionTime.IsZero()
+	return j.Status.CompletionTime != nil
+}
+
+// TODO: Turn this into an utility in pjutil?
```
Comment (spxtr) [inline prow/kube/prowjob.go]: Ditto.

**Comment by spxtr** [comment]
LGTM

**Pull request merged**